pkg/kubelet/kubelet_pods.go

```diff
@@ -1171,6 +1188,32 @@ func (kl *Kubelet) HandlePodCleanups() error {
 	}
 
 	kl.backOff.GC()
+
+	// If two pods with the same UID are observed in rapid succession, we need to
+	// resynchronize the pod worker after the first pod completes and decide whether
+	// to restart the pod. This happens last to avoid confusing the desired state
+	// in other components and to increase the likelihood transient OS failures during
+	// container start are mitigated. In general only static pods will ever reuse UIDs
+	// since the apiserver uses randomly generated UUIDv4 UIDs with a very low
+	// probability of collision.
+	for uid := range restartablePods {
+		pod, ok := allPodsByUID[uid]
```

Comment:
i am trying to work this logic backward.  in this scenario, we have `workingPods` list that contains two discrete pods with the same uid, right?  we then map pods by uid in the `allPodsByUID` list, but it's not clear which pod in the `workingPods` list wins in that mapping, right?  am i missing something?

Reply:
ignore my comment here, the HandlePodAdditions step will result in the most recent static pod being present in the pod manager for that uid.